vm-setup/roles/v1aX_integration_test/tasks/remediation.yml

```diff
@@ -1,47 +1,45 @@
 ---
-  - name: Get the name of a worker bmh name
-    shell: |
-       kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
-       |select(.status.provisioning.state | contains("provisioned")) |select(.spec.consumerRef.name
-       | contains("{{ CLUSTER_NAME }}-workers")) | .metadata.name'
-    register: worker_bmh_name
+  - name: Get the names of worker BareMetalHosts
+    shell: kubectl get m3m -n "{{ NAMESPACE }}" -o json  | jq -r '.items[] | select(.metadata.name | contains("workers")) |.metadata.annotations."metal3.io/BareMetalHost"'
```

Comment:
If we fetch the BMH name from BMH and machine name from machines or M3M object, the order of the bmh name and corresponding Machine name are not same always . That is the reason, the feature test for remediation were failing.  Here in this solution, Feruz tried to fetch both BMH name and machine name from M3M object, so that the order of the BMH name and machines stays same.

Reply:
With the current changes, we are doing only 2 API calls. One for fetching worker BMH & Machine and second for master BMHs and Machines. We are using Machine names as k8s node name here. The reason for fetching BMH from m3m is to be able to find correct BMH that is served by some particular Machine object. Because, for example when we do `kubectl get machines ` -  the order of the output of the Machines is not the same with the order of the output of the `kubectl get bmh`. Meaning that, it isn't always 1-1 mapping, sometimes the order is different either in Machines or BMHs, and that is causing wrong correlation between BMH and Machine.